fission/src/systems/scene/SceneRenderer.ts

```diff
@@ -328,83 +339,11 @@ class SceneRenderer extends WorldSystem {
-            (event: { target: TransformControls; value: unknown }) => {
-                const isAnyGizmoDragging = Array.from(this._transformControls.keys()).some(gizmo => gizmo.dragging)
-                if (!event.value && !isAnyGizmoDragging) {
-                    this._orbitControls.enabled = true // enable orbit controls when not dragging another transform gizmo
-                } else if (!event.value && isAnyGizmoDragging) {
-                    this._orbitControls.enabled = false // disable orbit controls when dragging another transform gizmo
-                } else {
-                    this._orbitControls.enabled = !event.value // disable orbit controls when dragging transform gizmo
-                }
-
-                if (event.target.mode === "translate") {
-                    this._transformControls.forEach((_size, tc) => {
-                        // disable other transform gizmos when translating
-                        if (tc.object === event.target.object && tc.mode !== "translate") {
-                            tc.dragging = false
-                            tc.enabled = !event.value
-                            return
-                        }
-                    })
-                } else if (
-                    event.target.mode === "scale" &&
-                    (InputSystem.isKeyPressed("ShiftRight") || InputSystem.isKeyPressed("ShiftLeft"))
-                ) {
-                    // scale uniformly if shift is pressed
-                    transformControl.axis = "XYZE"
-                } else if (event.target.mode === "rotate") {
-                    // scale on all axes
-                    this._transformControls.forEach((_size, tc) => {
-                        // disable scale transform gizmo when scaling
-                        if (tc.mode === "scale" && tc !== event.target && tc.object === event.target.object) {
-                            tc.dragging = false
-                            tc.enabled = !event.value
-                            return
-                        }
-                    })
-                }
-            }
-        )
-
-        this._transformControls.set(transformControl, size)
-        this._scene.add(transformControl)
-
-        return transformControl
-    }
-
-    /**
-     * Remove transform gizmos from Mesh
-     *
-     * @param obj Mesh to remove gizmo from
-     * @returns void
-     */
-    public RemoveTransformGizmos(obj: THREE.Object3D) {
-        this._transformControls.forEach((_, tc) => {
-            if (tc.object === obj) {
-                tc.detach()
-                this._scene.remove(tc)
-                this._transformControls.delete(tc)
-            }
-        })
+    /** returns whether any gizmos are being currently dragged */
+    public IsAnyGizmoDragging(): boolean {
+        return Array.from(this._sceneObjects.values())
+            .filter(obj => obj instanceof GizmoSceneObject)
+            .some(obj => obj.gizmo.dragging)
     }
```

Comment:
```suggestion
    public IsAnyGizmoDragging(): boolean {
        return Array.from(this._gizmosOnMirabuf.values())
            .some(gizmo => gizmo.dragging)
    }
```

I'm not sure if I got the syntax/names right but you can skip filtering since you're already keeping a map of active gizmos.